ffcx/ir/dof_permutations.py

```diff
@@ -200,28 +207,84 @@ def face_tangents_from_subdofmap(ufl_element):
     dof_types = [e.functional_type for e in dual]
     entity_dofs = fiat_element.entity_dofs()
 
-    rotations = []
-    if cname == "tetrahedron":
+    face_tangents = {}
+    if cname == "tetrahedron" or cname == "hexahedron":
         # Iterate through faces
         for entity_n in range(len(entity_dofs[2])):
             dofs = entity_dofs[2][entity_n]
             types = [dof_types[i] for i in dofs]
+            if cname == "tetrahedron":
+                tangent_data = {i: {} for i in range(6)}
+            elif cname == "hexahedron":
+                tangent_data = {i: {} for i in range(8)}
 
             # PointFaceTangent dofs
-            if "PointFaceTangent" in types:
+            if cname == "tetrahedron" and "PointFaceTangent" in types:
                 type_dofs = [i for i, t in zip(dofs, types) if t == "PointFaceTangent"]
                 for dof_pair in zip(type_dofs[::2], type_dofs[1::2]):
-                    # (entity_dim, entity_number), dofs
-                    rotations.append(((2, entity_n), dof_pair))
+                    # rotations[(entity_dim, entity_number)][face_permuation][dof] = [(d_i, a_i)]
```

Comment:
Leftover commented code?

Reply:
It was a comment to myself to remind me how I've stored this data. I've moved it to the top of the function with more words